src/models/inbound/dfspConsentRequests.model.ts

```
@@ -0,0 +1,243 @@
+      const response = await this.dfspBackendRequests.validateConsentRequests(request)
+
+      if (!response) {
+        throw new Errors.MojaloopFSPIOPError('', '', '', Errors.MojaloopApiErrorCodes.TP_FSP_CONSENT_SCOPES_ERROR)
+      }
+
+      if (!response.isValid) {
+        throw new Errors.MojaloopFSPIOPError('', '', '', Errors.MojaloopApiErrorCodes.TP_NO_SUPPORTED_SCOPE_ACTIONS)
+      }
+
+      this.data.response = response
+
+      type consentRequestResponseType = tpAPI.Schemas.ConsentRequestsIDPutResponseOTP &
+        tpAPI.Schemas.ConsentRequestsIDPutResponseWeb
+      const consentRequestResponse = {
+        scopes: request.scopes,
+        callbackUri: request.callbackUri,
+        authChannels: response.authChannels,
+        authUri: response.authUri,
+        initiatorId: toParticipantId
+      } as consentRequestResponseType
+      await this.thirdpartyRequests.putConsentRequests(request.id, consentRequestResponse, toParticipantId)
+
+
+    } catch (error) {
+      const mojaloopError = await reformatError(error)
+
+      this.logger.push({ error }).error('start -> RequestIsValid')
+      this.logger.push({ mojaloopError }).info(`Sending error response to ${toParticipantId}`)
+
+      await this.thirdpartyRequests.putConsentRequestsError(
+        request.id,
+        mojaloopError as unknown as fspiopAPI.Schemas.ErrorInformationObject,
+        toParticipantId
+      )
+
+      // throw error to stop state machine
+      throw error
+    }
+  }
+
+  async onStoreReqAndSendOTP (): Promise<void> {
+    const { request, response } = this.data
+
+    try {
+      const channel = [...response!.authChannels].pop()
+      switch (channel) {
+        case 'WEB': {
+          await this.dfspBackendRequests.storeConsentRequests(request)
+          break
+        }
+        case 'OTP': {
+          await this.dfspBackendRequests.sendOTP(request)
+          break
+        }
+        default: {
+          console.error(`Invalid authChannel ${channel}`)
```

I really think this should throw an error here.

---

We probably also shouldn't log to console... we have a logger, right?

---

ok done, instead of just logging, throwing new error in case of invalid auth channel

---

cool